packages/ember-glimmer/tests/integration/helpers/yield-test.js

```diff
@@ -0,0 +1,256 @@
+    let KiwiCompComponent = Component.extend({
+      boundText: 'inner'
+    });
+
+    this.registerComponent('kiwi-comp', { ComponentClass: KiwiCompComponent, template: '<p>{{boundText}}</p><p>{{yield}}</p>' });
+
+    this.render('{{#with boundText as |item|}}{{#kiwi-comp}}{{item}}{{/kiwi-comp}}{{/with}}', { boundText: 'outer' });
+
+    equal(this.$('div p:contains(inner) + p:contains(outer)').length, 1, 'inner component property isn\'t masked by outer keyword on render');
+    let element1 = this.firstChild;
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'boundText', 'otherworld'));
+
+    equal(this.$('div p:contains(inner) + p:contains(otherworld)').length, 1, 'inner component property isn\'t masked by outer keyword when rerendered with new context');
+
+    this.runTask(() => set(this.context, 'boundText', 'outer'));
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@test inner keyword doesn\'t mask yield property']() {
+    let KiwiCompComponent = Component.extend({
+      boundText: 'inner'
+    });
+
+    this.registerComponent('kiwi-comp', { ComponentClass: KiwiCompComponent, template: '{{#with boundText as |item|}}<p>{{item}}</p><p>{{yield}}</p>{{/with}}' });
+
+    this.render('{{#kiwi-comp}}{{item}}{{/kiwi-comp}}', { item: 'outer' });
+
+    equal(this.$('div p:contains(inner) + p:contains(outer)').length, 1, 'outer property isn\'t masked by inner keyword on render');
+
+    let element1 = this.firstChild;
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'item', 'otherworld'));
+
+    equal(this.$('div p:contains(inner) + p:contains(otherworld)').length, 1, 'inner component property isn\'t masked by outer keyword when rerendered with new context');
+
+    this.runTask(() => set(this.context, 'item', 'outer'));
+
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@test can bind a keyword to a component and use it in yield']() {
+    let KiwiCompComponent = Component.extend({
+      boundText: 'inner'
+    });
+
+    this.registerComponent('kiwi-comp', { ComponentClass: KiwiCompComponent, template: '<p>{{content}}</p><p>{{yield}}</p>' });
+
+    this.render('{{#with boundText as |item|}}{{#kiwi-comp content=item}}{{item}}{{/kiwi-comp}}{{/with}}', { boundText: 'outer' });
+
+    equal(this.$('div p:contains(outer) + p:contains(outer)').length, 1, 'component and yield have keyword');
+
+    let element1 = this.firstChild;
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'boundText', 'update'));
+
+    equal(this.$('div p:contains(update) + p:contains(update)').length, 1, 'keyword has correctly propagated update');
+
+    this.runTask(() => set(this.context, 'boundText', 'outer'));
+
+    let element2 = this.firstChild;
+
+    this.assertSameNode(element1, element2);
+  }
+
+  ['@htmlbars yield view should be a virtual view']() {
```

Comment:
"virtual view" is no longer terminology in Ember. `yield should not introduce a view` perhaps?